### PR TITLE
Favour `self.var = _` over `@var = _`

### PR DESCRIPTION
**Pull request by AndrewMcBurney**
Unify instance variable setters and getters by favouring `self.var = _` unless memoized `@instance` variable encapsulated in method.

**Changes**

- [x] Favour `self.var = _` over `@var = _` for setting instance variables
- [x] Unify memoized methods:

```ruby
def var_name
  @var_name ||= _
end
```

- [x] User `attr_accessor :var_name` over:

```ruby
def var_name
  return @var_name
end
```

### Todo

- [x] Make sure nothing broke 😂

### CODE REVIEW

File: features/build_runner/build_runner.rb [hunk 1]
```diff
@@ -41,16 +41,16 @@ class BuildRunner
     def initialize(project:, sha:, github_service:, work_queue: nil)
       # Setting the variables directly (only having `attr_reader`) as they're immutable
       # Once you define a FastlaneBuildRunner, you shouldn't be able to modify them
-      @project = project
-      @sha = sha
+      self.project = project
```
[the remark this answers is not in the document]
Comment (snatchev) [reply features/build_runner/build_runner.rb]: 👍
I think this a good, generally-accepted approach!

[the remark this answers is not in the document]
Comment (taquitos) [reply features/build_runner/build_runner.rb]: I don't know about `def mutate_var!(val = "Mutated value")`, that seems like we'll need to add an extra method anytime we want to have a variable mutate. I think we should just stick to using `object.var = `

[the remark this answers is not in the document]
Comment (AndrewMcBurney) [reply features/build_runner/build_runner.rb]: Oh, sorry - that was just an example usecase. I didn't really make it clear, my apologies. 😳 I meant we should use `self.var = val` like you said.

The `mutate_var!` method was just an example showcasing the internals of how we'd mutate variables (i.e., with `self.var = val`)

[the remark this answers is not in the document]
Comment (AndrewMcBurney) [reply features/build_runner/build_runner.rb]: Later this evening, I'll update the PR to use the approach described above. 💯

[the remark this answers is not in the document]
Comment (taquitos) [reply features/build_runner/build_runner.rb]: Oh, duh, yeah, I was being dense. Sounds awesome!

**Comment by taquitos** [comment]
Still love this idea, but it's out of sync with master after we moved a bunch of files. Closing for now.

**Pull request closed without merging**